### PR TITLE
Improve closing study workflow

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?
There was a bit a of a race condition when setting the lock to a study-card after closing the study, specially when running in not so powerful machines. This PR wants to fix that. Let's see.

e2e: Check study state before deleting

## Related issue number
related to https://github.com/ITISFoundation/osparc-simcore/issues/1987

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: tests/e2e/tutorials/tutorialBase.js [hunk 1]
```diff
@@ -303,11 +319,10 @@ class TutorialBase {
     await this.takeScreenshot("closeStudy_after");
   }
 
-  async removeStudy() {
+  async removeStudy(studyId) {
     await this.takeScreenshot("deleteFirstStudy_before");
     try {
-      // wait for projects to be loaded
-      await utils.sleep(2000);
+      await this.waitForStudyUnlocked(studyId);
```
Comment (pcrespov) [inline tests/e2e/tutorials/tutorialBase.js:325]: why dont you pass the former sleep time (i.e. 2000) as the timeout of ``waitForStudyUnlocked``?

File: tests/e2e/tutorials/tutorialBase.js [hunk 2]
```diff
@@ -225,6 +225,18 @@ class TutorialBase {
     return;
   }
 
+  async waitForStudyUnlocked(studyId, timeout = 10000) {
+    const start = new Date().getTime();
+    while ((new Date().getTime())-start < timeout) {
+      await utils.sleep(1000);
```
Comment (pcrespov) [inline tests/e2e/tutorials/tutorialBase.js:231]: the ``sleep`` here will produce a timeout if the value of ``timeout`` is smaller that ``1000``
do something like `` timeout/100``

**Review by pcrespov — CHANGES_REQUESTED** [review]
Much better

File: tests/e2e/tutorials/tutorialBase.js (hunk 1 above)
Comment (odeimaiz) [reply tests/e2e/tutorials/tutorialBase.js:325]: Because it can take a bit longer closing the dynamic services. The default is 10".

**Review by pcrespov — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]
great!

**Pull request merged**